### PR TITLE
Label prs features

### PR DESCRIPTION
**Pull request by idelahoz**
There are two main additions on this PR:
##### `code ready`

It marks the current branch PR as "awaiting review", the label has to be create previously.
##### `code pr`

When the current branch is a hotfix we generate two PRs. We mark those 2 PRs with the "hotfix" label.
The "hotfix" label also has to be created previously.

### CODE REVIEW

File: bin/code [hunk 1]
```diff
@@ -44,7 +57,13 @@ command :publish, :pr do |c|
     base_branch = nil
     base_branch = Code::Branch.master if options[:master]
 
-    $git.publish(base: base_branch, message: message)
+    github_api = Code::GitHubAPI.new
+
+    if (github_api.current_branch_pr?)
```
Comment (venkatd) [inline bin/code]: What does `current_branch_pr` mean? Can you write this block of code in the what-do-I-want style?

File: bin/code [hunk 2]
```diff
@@ -55,6 +74,14 @@ command :push, :p do |c|
   end
 end
 
+desc "Marks the current branch's pr as awaiting review"
+command :ready do |c|
+  c.action do |global_options, options, args|
+     github_api = Code::GitHubAPI.new
+     github_api.mark_current_pr_as_awaiting_review
```
Comment (venkatd) [inline bin/code]: Don't you want want a `PullRequest` object? That's what you _want_ isn't it? (But you didn't have one.)

File: bin/code [hunk 3]
```diff
@@ -43,8 +56,13 @@ command :publish, :pr do |c|
 
     base_branch = nil
     base_branch = Code::Branch.master if options[:master]
+    current_branch = Code::Branch.current
 
-    $git.publish(base: base_branch, message: message)
+    if (current_branch.has_pr?)
+      Code::System.open_in_browser(current_branch.pull_request_url)
```
Comment (idelahoz) [inline bin/code:62]: @begedin I had to refactor this for @venkatd corrections on this PR, before working on your branch take a look at this and merge it if it seems correct so we avoid conflicts.

Basically, I made the `Code::Branch` instance the main object to interact on the higher level. So the branch object knows the PR related to it `branch.pull_request` or `branch.pull_requests` if you need all the list.

File: lib/code/pull_request.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    def self.fetch_prs_for_branch(branch)
+      github_api.prs_for_branch(branch)
+    end
+    
+    # I need a github api instance to create instances
+    # and also need an api instance once the api is created
+    # I tried to solve this problem with modules and complicated
+    # solutions, but I'll duplicate it for now.
+    def self.github_api
```
Comment (idelahoz) [inline lib/code/pull_request.rb:44]: @venkatd I did some research but didn't find a good answer for this. I duplicated the method on instance and class level, but it it has a bad smell.

File: lib/code/github_api.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        token = authorization[:token]
+      else
+        token = client.create_authorization(scopes: ["repo"], note: AUTHORIZATION_NOTE)[:token]
+      end
+
+      puts "Successfully authorized with token #{token}"
+      self.authorization_token = token
+    end
+
+    def prs_for_branch(branch)
```
Comment (idelahoz) [inline lib/code/github_api.rb]: This method replaces the 4 methods that were here before(current_branch_pr, current_branch_pr?, current_branch_prs and current_branch_pr_url) that didn't really belong here. That responsibility is now for Branch class and PullRequest class.

**Comment by idelahoz** [comment]
@venkatd I refactored the code, tried hard to stick to the principles. Take a look.

File: lib/code/github_api.rb (hunk 5 above)
Comment (venkatd) [inline lib/code/github_api.rb]: Hey I'm thinking better to spell it out completely as in `pull_requests`

Comment (venkatd) [inline lib/code/github_api.rb]: For all the code.

File: bin/code (hunk 3 above)
Comment (venkatd) [inline bin/code:62]: Hey @idelahoz this is awesome! Great job. Seriously. If you keep thinking about the code as what-do-I-want you'll usually write great code naturally.

**Comment by venkatd** [comment]
Hey @idelahoz everything looks great. Good job with the pull request object and thinking top down. Small change is replacing `pr` with `pull_request`.

**Comment by begedin** [comment]
@idelahoz @venkatd I did the method rename myself. Since this pr contains most of the stuff in pull request #9, and there was re-factoring done to it, I think it might actually be easier to just merge this one for both features and close #9. I'll take a look to see if everything is contained here and add modifications if necessary.

Merging the other one first would be a nightmare in comparison, in my opinion.

**Comment by begedin** [comment]
@idelahoz @venkatd I actually managed to apply changes so that it's possible to merge this one, followed by merging #9, so that's what I'll do. 

Since @venkatd said all that's left here are method renames, I'll assume I can merge this without waiting, now that I did the renames myself.

**Comment by begedin** [comment]
@idelahoz @venkatd 

Disregard the previous comment. I didn't think this through fully. Merging this one means the open pr feature is now also merged.

What's left on my side is to rewrite the OS detecting code and possibly add an instance variable for the client in github_api, so I'll create a new branch/pr for that.

**Pull request merged**